### PR TITLE
feat: [CI-10684]: CI gitness integration with new UX

### PR DESCRIPTION
**Pull request by karansaraswat19**
#### Screenshots

Screenshot 2023-12-10 at 8 00 53 PM (2)
Screenshot 2023-12-10 at 8 01 07 PM (2)
Screenshot 2023-12-10 at 8 01 16 PM (2)

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by karansaraswat19** [comment]
retrigger jest

**Review by vardanbansal-harness — DISMISSED** [review]

File: src/modules/10-common/components/GitProviderSelect/__tests__/GitProviderSelect.test.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          gitProvider={getGitProviderCards(getString)[0]}
+          setFieldValue={jest.fn()}
+          connectorFieldName="connectorRef"
+          repoNameFieldName="repoName"
+          handleChange={jest.fn()}
+          showDescription={true}
+        />
+      </TestWrapper>
+    )
+    expect(container).toMatchSnapshot()
```
Comment (ishantmehtaharness) [inline src/modules/10-common/components/GitProviderSelect/__tests__/GitProviderSelect.test.tsx:24]: We should avoid snapshot testing as much as possible
cc: @chetanvekariya-harness

File: src/modules/70-pipeline/components/PipelineStudio/CloneCodebaseForm/__tests__/CloneCodebaseForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    expect(getConnectorWidth({ connectorWidth: 460, connectorRef: 'gitsync2' })).toBe(460)
+    expect(getConnectorWidth({ connectorWidth: 460, connectorRef: '<+input>' })).toBe(402)
+    expect(getConnectorWidth({ connectorRef: '<+input>' })).toBe(undefined)
+  })
+})
+
+describe('CloneCodebaseForm tests', () => {
+  test('initial render with ff on', async () => {
+    const { container, getByText } = render(<WrapperComponent isCodeEnabled />)
+    expect(container).toMatchSnapshot()
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/PipelineStudio/CloneCodebaseForm/__tests__/CloneCodebaseForm.test.tsx:96]: https://github.com/harness/harness-core-ui/pull/19348/files#r1426358467

**Review by shaurya-harness — APPROVED** [review]

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]
You can avoid writing test cases and add proper test cases later, if these changes are urgently required. Let's not add new snapshots.

File: src/modules/10-common/components/GitProviderSelect/__tests__/GitProviderSelect.test.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/10-common/components/GitProviderSelect/__tests__/GitProviderSelect.test.tsx:24]: Thanks, @ishantmehtaharness . That is correct.

Comment (chetanvekariya-harness) [reply src/modules/10-common/components/GitProviderSelect/__tests__/GitProviderSelect.test.tsx:24]: @karansaraswat19 - Remove `expect(container).toMatchSnapshot()`

**Comment by karansaraswat19** [comment]
@chetanvekariya-harness @ishantmehtaharness will remove the snapshots in a follow-up PR

**Review by ishantmehtaharness — APPROVED** [review]
If this is urgent, then lets merge. Else we should add proper jest tests, rather than snapshots

**Comment by ishantmehtaharness** [comment]
Coverage, Eslint failing

**Review by chetanvekariya-harness — APPROVED** [review]
As discussed, with @karansaraswat19 , he will be removing snapshots in the next PR which will be raised immediately after this one. So, approving just because I had blocked this PR because of that. Meaningful, reviews to be taken from CI team.

**Pull request merged**